### DISCUSSION TITLE
Multiple User Support

### DISCUSSION THREAD

**Discussion by creuzerm** [ItzCrazyKns/Vane#230] [Ideas]
I just got this tool working today after a few false starts the last couple of weeks. (Trying to install into Portainer was more than I could figure out, so I ended up with a clean Proxmox Container with just Docker on it with success after some extra configuration not on the quick start screen)

I like how we have the history. I do want to support multiple users like what we have with OpenWebUI. 

I am thinking a quick and dirty way would be to deploy it multiple times. For maintenance, i think having 1 searxng and multiple sets of Perplexica would be more convenient than entire duplicate deployments because then I can update once.

I'd rather see more time improving the things that I can't fake. Anybody else have ideas on how we can support multiple users easily?

**Comment by CSAnetGmbH** [comment]
Thats a great idea !

**Comment by poyasyn** [comment]
would be the best feature for family use cases for sure !

**Comment by Avuja** [comment]
Yeah adding on to this if we could obfuscate the API keys that'd be nice too. I'm standing up a deployment of Perplexica for myself and some friends to replace Perplexity but currently with no login/auth/users it also seemingly is presenting any API keys configured in plaintext via the Settings panel

**Comment by delfireinoso** [comment]
Another thing that it's near to be seen: Login/Auth/Users. Perplexica for the family and friends and no keys on sight

**Comment by cedhuf** [comment]
Hey everyone! 
Just deployed Vane today and I'm really impressed with the app !

That said, I'm running into the exact same issue as OP — I want to share this with family members, and a shared search history is a non-starter from a privacy standpoint.

I see a few possible approaches:
- Native multi-user support with isolated history per account (ideal)
- OIDC/OAuth2 authentication so identity providers like PocketID can handle access control
- At minimum, a way to scope history to a browser session rather than the server — no user accounts needed, just isolate history client-side (localStorage?) so each browser/device has its own independent history. Not perfect, but would already be a huge improvement for family use cases without requiring a full auth system.

Any update on plans for this? The project has come a long way — multi-user support feels like the last missing piece for family/household deployments.

Thanks for the amazing work! 🙏

Edit: I just came across PR #1107 which seems to address exactly this — fingers crossed it gets merged soon!
